### PR TITLE
New addon: Scratch 3.0 → 2.0

### LINKED ISSUE #985: 3.0 → 2.0
**Describe the new addon**
So there is an addon called "2.0 → 3.0" so, I decided to come up with a 3.0 → 2.0! This would turn any 3.0 pages into 2.0 pages!

**Link to the new addon**
Every 3.0 Page. scratch.mit.edu scratch.mit.edu/projects/xxxxxxxx/ (not the editor or the project player though) and every other 3.0 styled page.

### LINKED ISSUE #7705: Old project layout
### Why this would be helpful

Does this remind you of anything?

300px-Scratch_2_project_page

We should add something like this that makes the project page look like it, we done that with studios already.

### How this addon works

Changes the Scratch 3.0 project page's layout to look more like Scratch 2's version.

### Possible alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mxmou**
Resolves #985
Resolves #7705

An addon that changes scratch-www pages to look like scratchr2. Less commonly used pages can be added in later PRs. In the screenshots below, old studio layout is enabled but the 2.0 preset in website dark mode isn't.

image
image
image

### CODE REVIEW

**Comment by WorldLanguages** [comment]
I guess this can't be enabled simultaneously with `2.0 → 3.0`

**Comment by mxmou** [comment]
> I guess this can't be enabled simultaneously with `2.0 → 3.0`

It can be: https://github.com/ScratchAddons/ScratchAddons/issues/985#issuecomment-744082987

I don't know how a Scratch 1.x theme, if it's ever added, could interact with the existing ones though.

**Comment by rickbot-dot** [comment]
That's a mouthful of commits :thinking:

**Comment by medlans** [comment]
Nice, also there is a similar userstyle:
https://github.com/mrbatguy0314/Scratch-3-to-Scratch-2
https://userstyles.world/style/11602/scratch-3-to-2
Also, I can give you some of mine I guess LOL

**Comment by medlans** [comment]
> I guess this can't be enabled simultaneously with `2.0 → 3.0`

I mean, the existing scratchr2 userstyle can LOL (it will not let me edit soo)

**Comment by WorldLanguages** [comment]
How does this play with "old studio layout"?

**Comment by WorldLanguages** [comment]
> I think old studio layout should stay separate, but its mode setting could be removed

Yes, that's up to you, the setting can be removed.

**Comment by SifaOrange** [comment]
There should really be an accurate scratch 2.0 userstyle.

File: addons/scratch3to2/addon.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  "credits": [
+    {
+      "name": "Maximouse",
+      "link": "https://scratch.mit.edu/users/Maximouse/"
+    }
+  ],
+  "dynamicEnable": true,
+  "dynamicDisable": true,
+  "injectAsStyleElt": true,
+  "versionAdded": "1.39.0",
```
Comment (WorldLanguages) [inline addons/scratch3to2/addon.json:14]:
```suggestion
  "versionAdded": "1.40.0",
```

**Review by WorldLanguages — APPROVED** [review]

**Comment by WorldLanguages** [comment]
It's funny how this addon is the opposite of the userstyle that inspired Scratch Addons.

**Comment by CST1229** [comment]
> It's funny how this addon is the opposite of the userstyle that inspired Scratch Addons.

Evil Scratch Addons be like: Scratch 3.0 -> 2.0

**Comment by medlans** [comment]
> Nice, also there is a similar userstyle: https://github.com/mrbatguy0314/Scratch-3-to-Scratch-2 https://userstyles.world/style/11602/scratch-3-to-2 Also, I can give you some of mine I guess LOL

This one changes a lot of stuff by the way (though it hides the extensions on the project page which also hides the cloud data thing, and the pages that it really works for are studios, projects, the main page, you can use some of its styles)

**Comment by WorldLanguages** [comment]
I can merge as admin on request.

**Comment by Samq64** [comment]
`dark-www`'s message count color, input background, and colored area settings seem to be overridden when `scratch3to2` is enabled.

**Comment by mxmou** [comment]
> `dark-www`'s message count color, input background, and colored area settings seem to be overridden when `scratch3to2` is enabled.

I'll fix the message count - I started working on this addon before the setting was added. I don't think input background and dark colored areas have any effect on vanilla scratchr2 pages, which this addon tries to imitate (inputs are controlled by the content background color).

**Comment by VojtaBubela11** [comment]
how do i add the addon to my browser?

(Chrome Browser)

**Comment by mxmou** [comment]
> how do i add the addon to my browser?
> 
> (Chrome Browser)

You can download it on [this page](https://github.com/mxmou/ScratchAddons/tree/scratch3to2): click Code, then Download ZIP. Extract the zip file and follow [these instructions](https://scratchaddons.com/docs/getting-started/installing/#installing-on-google-chrome-or-microsoft-edge) to add it to the browser. If you already have Scratch Addons installed, you need to disable that version because having two versions of the same extension at once will cause bugs.

**Comment by WorldLanguages** [comment]
@mxmou Do we want this one for 1.41.0 ?

File: addons/scratch3to2/addon.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  "credits": [
+    {
+      "name": "Maximouse",
+      "link": "https://scratch.mit.edu/users/Maximouse/"
+    }
+  ],
+  "dynamicEnable": true,
+  "dynamicDisable": true,
+  "injectAsStyleElt": true,
+  "versionAdded": "1.40.0",
```
Comment (WorldLanguages) [inline addons/scratch3to2/addon.json:14]:
```suggestion
  "versionAdded": "1.41.0",
```

File: addons/scratch3to2/addon.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
+{
+  "name": "Scratch 3.0 \u2192 2.0",
+  "description": "Makes Scratch 3.0-styled pages look like Scratch 2.0.",
+  "tags": ["community", "theme"],
```
Comment (WorldLanguages) [inline addons/scratch3to2/addon.json:4]:
```suggestion
  "tags": ["community", "theme", "featured"],
```

**Comment by mxmou** [comment]
> @mxmou Do we want this one for 1.41.0 ?

It can be merged once I update it to support the new Ideas and CGs, which shouldn't take long.

**Comment by mxmou** [comment]
I think this is ready to be merged. It doesn't style every page yet, but it doesn't break anything and we can add support for less commonly visited pages in later updates.

**Comment by WorldLanguages** [comment]
Sounds good to me

**Comment by CST1229** [comment]
> I think this is ready to be merged. It doesn't style every page yet, but it doesn't break anything and we can add support for less commonly visited pages in later updates.

beta tag?

**Comment by mxmou** [comment]
> beta tag?

I don't think it's needed - the addon doesn't break anything, it just keeps the 3.0 design mostly unchanged on some pages.

**Comment by Secret-chest** [comment]
Wait, aren't the photos actually 2.0? Good job!

**Comment by mxmou** [comment]
> Wait, aren't the photos actually 2.0? Good job!

They're 3.0 with the addon! Old studio layout is also enabled in the studio screenshot.

**Comment by SifaIssac** [comment]
we do need the editor and fullscreen player though

**Comment by DNin01** [comment]
> we do need the editor and fullscreen player though

There are already addons that you can set up to make the editor look more like 2.0.

**Comment by CST1229** [comment]
> There are already addons that you can set up to make the editor look more like 2.0.

the need for superpresets is bigger than ever

**Comment by SifaIssac** [comment]
> > There are already addons that you can set up to make the editor look more like 2.0.
> 
> the need for superpresets is bigger than ever

which is the 2.0 blocks syle,bevel,icons, and the font

**Comment by WorldLanguages** [comment]
In the Discord server, 1 person asked for a setting to disable the 2.0 icons (the feedback message has no more detail, but I imagine they mean the 2.0 views icon in project pages, etc.)

**Comment by mxmou** [comment]
> In the Discord server, 1 person asked for a setting to disable the 2.0 icons (the feedback message has no more detail, but I imagine they mean the 2.0 views icon in project pages, etc.)

A 2.0 theme with 3.0 icons would look like a mess. It would make sense if there was a specific reason for someone to prefer the new icons, but I can't think of one.

**Comment by MaritoMaestro** [comment]
this time, projects requires Flash Player??

**Comment by CST1229** [comment]
> this time, projects requires Flash Player??

no

**Comment by MaritoMaestro** [comment]
> > this time, projects requires Flash Player??
> 
> no

why??

**Comment by CST1229** [comment]
> > > this time, projects requires Flash Player??
> > 
> > 
> > no
> 
> why??

because it does not literally go back in time to 2.0, it just makes the 3.0 pages *look* like 2.0. 2.0 -> 3.0 does the exact same thing

**Comment by MaritoMaestro** [comment]
What Styles uses the See Inside and See Project Page Buttons?? 2.0 or 3.0??

**Comment by CST1229** [comment]
> What Styles uses the See Inside and See Project Page Buttons?? 2.0 or 3.0??

enable the addon and see for yourself. it's in 1.42

**Comment by MaritoMaestro** [comment]
See Inside is 2.0 and See Project Page is 3.0??

**Comment by MaritoMaestro** [comment]
this time, projects require WebGL instead of Flash Player??

**Comment by Samq64** [comment]
No functionally is added or removed. The addon is categorized as a website theme, which means it only changes the **style** of the Scratch website which is generally considered separate from the project editor.

**Comment by MaritoMaestro** [comment]
On the ideas page you see 3.0 images on the 2.0 theme??

**Pull request merged**